DOC Add link to plot_lasso_and_elasticnet.py example in linear model

marouaneaka opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
<!--
Towards 26927
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.
Towards 26927
Added links to lasso and elasticnet classes

#### Any other comments?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
https://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

Commit:
DOC Add link to plot_lasso_and_elasticnet.py example in linear model lasso and elastic net

Commit:
DOC Add link to plot_lasso_and_elasticnet.py example in linear model lasso and elastic net

Commit:
DOC Add link to plot_lasso_and_elasticnet.py example in linear model lasso and elastic net patched

Commit:
DOC Add link to plot_lasso_and_elasticnet.py example in linear model lasso and elastic net

marouaneaka committed:
Merge branch 'scikit-learn:main' into update-docs

Commit:
DOC Add link to plot_lasso_and_elasticnet.py example in linear model lasso and elastic net

virchan commented on sklearn/linear_model/_coordinate_descent.py:
```diff
@@ -875,6 +875,10 @@ class ElasticNet(MultiOutputMixin, RegressorMixin, LinearModel):
     >>> print(regr.predict([[0, 0]]))
     [1.451...]
+
+    For an example showcasing Elastic-Net alongside Lasso and ARDRegression for sparse
```
"ARD regression", and likewise for Lasso.

virchan reviewed (commented):
Thank you for the PR @marouaneaka! I have a few comments:

1. It might be helpful to include a link to the example in the [ARD regression documentation](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.ARDRegression.html#sklearn.linear_model.ARDRegression) as well.
2. In the PR description, please update the text to "Towards 26927" to avoid automatically closing the meta-issue when the PR is merged.

Commit:
added doc to ard regression

marouaneaka commented:
Hi thanks for the comments : - Added link in ARD Regression, fixed typo mistake in elastic net and lasso. if there is anything else just let me know :)

marouaneaka committed:
Merge branch 'main' into update-docs

virchan reviewed (commented):
Thanks for updating the PR, @marouaneaka! I have a few more comments.

In the ARD Regression documentation, could you please move the following example:

<img width="1048" alt="Screenshot 2025-01-10 142444" src="https://github.com/user-attachments/assets/9ae71fda-b7a2-4af1-ac32-cfc54ac383cc" />


down to where your L1 example is? After that, you can delete the "Notes" section.

Commit:
ARD reg documentation, usage example moved next to comparaison example

Commit:
lint fixed line char limit

marouaneaka commented:
Just updated ARD documentation as specified

Commit:
fixed ref to ard reg example

Commit:
fixed ref to ard reg example lint fix

virchan reviewed (commented):
Thank you for updating the PR, @marouaneaka! Could you address the CI failure? If you click on "Details" under "scikit-learn.scikit-learn," you’ll find the error message(s).

marenwestermann commented:
Hi @marouaneaka! Would you like to continue working on this PR? Let us know if you need help.

StefanieSenger commented:
This PR had been superseded by #31425 and can now be closed, since the superseding PR got merged.
